modules/file-transfer/src/main/java/org/apache/ignite/internal/network/file/ChunkedFileReader.java

```diff
@@ -88,18 +81,12 @@ byte[] readNextChunk() throws IOException {
             throw new IOException("No more chunks to read");
         }
 
-        int toRead = (int) Math.min(chunkSize, length - offset);
-        byte[] data = new byte[toRead];
-        int read = stream.read(data);
-
-        if (read != toRead) {
-            throw new IOException("Failed to read chunk data from file: expected " + toRead + ", actual " + read + "]");
-        }
+        byte[] data = stream.readNBytes(chunkSize);
 
-        offset += toRead;
         nextChunkNumber++;
+        hasNextChunk = stream.available() > 0;
 
-        if (offset == length) {
+        if (stream.available() == 0) {
```

Comment:
I wonder why it's not a `!hasNextChunk`, but whatever.
I've been looking at this code for too long. If there are more problems, I probably won't notice them at this point